### PR TITLE
Added init_students procedure

### PR DESCRIPTION
**Pull request by Kubasinska**
1. Added enroll_to_school method (class person)
2. Added init_students procedure  (class MesaProton)
3. Added decide_conn_number (class MesaProton)
4. Added makeSchoolLinks (class person)

### CODE REVIEW

File: mesa/Person.py [hunk 1]
```diff
@@ -156,6 +156,15 @@ def makeParent_OffspringsLinks(self, asker):
             self.neighbors.get("offspring").add(asker)
             asker.neighbors.get("parent").add(self)
 
+    def makeSchoolLinks(self, asker):
```
Comment (mariopaolucci) [inline mesa/Person.py:159]: What about avoiding repeated lines and going with something like the snippet here?
```
        if not isinstance(asker, Iterable): asker = [asker]
        for person in asker:
                self.neighbors.get("school").add(person)
                person.neighbors.get("school").add(self)
```

PS: does it ever get called from a person?

File: mesa/Person.py [hunk 2]
```diff
@@ -206,11 +215,27 @@ def init_person(self): # person command
...
 
+    def enroll_to_school(self, level):
+        """
+        Given a level of education, this method chooses a school where to enroll the agent
+        and modifies my_school atribute in-place.
+        :param level: int, level of education to enroll
+        """
+        self.potential_school = [school for agent in self.neighbors["household"] for school in agent.my_school if school.education_level == level]
+        if self.potential_school:
+            self.my_school = self.m.rng.choice(self.potential_school)
```
Comment (mariopaolucci) [inline mesa/Person.py:231]: repeated line; move outside of if

File: mesa/Person.py [hunk 3]
```diff
@@ -206,11 +215,27 @@ def init_person(self): # person command
...
+        else:
+            self.potential_school = [x for x in self.m.schools if x.diploma_level == level]
+            self.my_school = self.m.rng.choice(self.potential_school)
+        self.my_school.my_students.add(self)
+
```
Comment (mariopaolucci) [inline mesa/Person.py:236]: don't we need to add the students links at this point?

File: mesa/mesaPROTON_OC.py [hunk 4]
```diff
@@ -607,10 +607,43 @@ def setup_schools(self):
...
+            start_age = row[0]
+            end_age = row[1]
+            pool = [x for x in self.schedule.agents if
+                    x.age() >= start_age and x.age() <= end_age and x.education_level == level - 1 and x.max_education_level >= level]
+            for agent in pool:
+                agent.enroll_to_school(level)
+        for school in self.schools:
+            conn = self.decide_conn_number(school.my_students, 15)
+            for student in school.my_students:
+                total_pool = school.my_students.difference({student})
```
Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:631]: Just use `remove`.

File: mesa/mesaPROTON_OC.py [hunk 5]
```diff
@@ -607,10 +607,43 @@ def setup_schools(self):
...
+                conn_pool = list(self.rng.choice(list(total_pool), conn, replace=False))
+                student.makeSchoolLinks(conn_pool)
+
+    def decide_conn_number(self, agents, max_lim):
```
Comment (mariopaolucci) [inline mesa/mesaPROTON_OC.py:635]: use `extra.at_most`

File: mesa/Person.py (hunk 1 above)
Comment (Kubasinska) [reply mesa/Person.py:159]: The proposed solution is better. But at the moment it is never called this procedure for two individual agents, so I will eliminate the condition.

File: mesa/Person.py (hunk 2 above)
Comment (Kubasinska) [reply mesa/Person.py:231]: Sorry, I don't get the point here, aan you be more specific.

File: mesa/Person.py (hunk 3 above)
Comment (Kubasinska) [reply mesa/Person.py:236]: Links are added later, when all students are already enrolled in schools.

from mesa/mesaPROTON_OC.py (630-634)
```
      for student in school.my_students:
                total_pool = school.my_students.difference({student})
                conn_pool = list(self.rng.choice(list(total_pool), conn, replace=False))
                student.makeSchoolLinks(conn_pool)
```

File: mesa/mesaPROTON_OC.py (hunk 4 above)
Comment (Kubasinska) [reply mesa/mesaPROTON_OC.py:631]: The "remove" method removes agents in-place, we cannot modify a set while it is iterated, for this reason I used the "difference" method.

File: mesa/mesaPROTON_OC.py (hunk 5 above)
Comment (Kubasinska) [reply mesa/mesaPROTON_OC.py:635]: I forgot about this static method.

**Pull request merged**